hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -1074,11 +1079,14 @@ public AbfsRestOperation read(final String path,
       ContextEncryptionAdapter contextEncryptionAdapter,
       TracingContext tracingContext) throws AzureBlobFileSystemException {
     final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
-    addCustomerProvidedKeyHeaders(requestHeaders);
 
     AbfsHttpHeader rangeHeader = new AbfsHttpHeader(RANGE,
         String.format("bytes=%d-%d", position, position + bufferLength - 1));
     requestHeaders.add(rangeHeader);
+    addEncryptionKeyRequestHeaders(path, requestHeaders, false,
+        contextEncryptionAdapter, tracingContext);
+    requestHeaders.add(new AbfsHttpHeader(RANGE,
```

Comment:
why is this going in here when like 1085 sets this header too?

Reply:
Seems to be outdated.
Caused by merge conflicts but it was fixed in the latest commit: 590a003048de696ff12490d87a2d6e6c2553b77d

More merge conflicts need to be resolved now.
Will take them up.